rpc/packetizer_test.go

```diff
@@ -0,0 +1,93 @@
+package rpc
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+func createPacketizerTestProtocol() *protocolHandler {
+	p := newProtocolHandler(nil)
+	p.registerProtocol(Protocol{
+		Name: "abc",
+		Methods: map[string]ServeHandlerDescription{
+			"hello": {
+				MakeArg: func() interface{} {
+					return nil
+				},
+				Handler: func(context.Context, interface{}) (interface{}, error) {
+					return nil, nil
+				},
+				MethodType: MethodCall,
+			},
+		},
+	})
+	return p
+}
+
+// TestPacketizerDecodeInvalidFrames makes sure that the packetizer
+// can handle invalid frames and skip over them.
+func TestPacketizerDecodeInvalidFrames(t *testing.T) {
+	// Encode a mix of valid and invalid frames.
+	v1 := []interface{}{MethodCall, 1, "abc.hello", new(interface{})}
+	iv1 := "some string"
+	iv2 := 53
+	v2 := []interface{}{MethodCall, 2, "abc.hello", new(interface{})}
+	iv3 := false
+	v3 := []interface{}{MethodCall, 3, "abc.hello", new(interface{})}
```

Comment:
Might as well test with `MethodNotify`, `MethodResponse`, and `MethodCancel`.

Reply:
Done.